### ISSUE TITLE
Redactor fails to remove empty p tag on save 

### ISSUE THREAD

**Issue by mmikkel** [craftcms/redactor#377]
### Description

I'm seeing some odd/unwanted behaviour when authors attempt to clear/empty Redactor fields.

If an author clears out a Redactor field by first selecting all the text (i.e. Cmd+A) and then hitting the backspace button, the field value stored is always NULL (which is expected/wanted behaviour).

However, _if a field contains multiple paragraphs_, and the author clears out the field without first selecting all the text – for example, by putting the caret after the last character and then holding down backspace until the field is empty – the value stored will be an empty `<p>` tag with a single linebreak, i.e. this:

```php
craft\redactor\FieldData#1
(
    [craft\redactor\FieldData:_pages] => null
    [craft\htmlfield\HtmlFieldData:_rawContent] => '<p><br /></p>

'
    [Twig\Markup:content] => '<p><br /></p>

'
    [Twig\Markup:charset] => 'UTF-8'
)
```

Visually, the Redactor field _appears_ empty after saving, and if the entry is re-saved again (without touching the field) the empty `<p>` appears to be removed. But, this isn't something that is easy for authors to catch, and results in a lot of `{% if entry.text %}` template conditionals evaluating to `true`, even if there isn't actually any content in the field.

### Steps to reproduce

1. Add multiple paragraphs of text to a Redactor field in an entry
2. Save the entry
3. Remove all the text in the Redactor field by placing the caret after the last character in the field, then holding down the backspace button
4. Re-save the entry
5. Confirm that the field value saved is an empty `<p>` tag with a single `<br/>` inside.

### Additional info

- Craft version: 3.7.36
- PHP version: 7.4
- Database driver & version: MySQL 5.7
- Plugins & versions: Redactor 2.10.5

**Comment by andris-sevcenko** [comment]
@mmikkel Is it saved with a `<br>` or `<br/>` inside?

**Comment by mmikkel** [comment]
Here's the relevant part of the payload for the `actions/entry-revisions/save-draft` request that triggers when the field is emptied – looks like it's saved as `<p><br/></p>`:

![CleanShot 2022-03-31 at 15 49 04](https://user-images.githubusercontent.com/298510/161070928-5920e60e-110f-4ab6-970c-f970a6a5d135.png)

**Comment by mmikkel** [comment]
Here's a quick screencast, if it helps:

https://user-images.githubusercontent.com/298510/161071315-86f1f491-f9ee-457f-a1a1-00c54fc2b8b7.mp4

**Comment by mmikkel** [comment]
I think it's due to some trailing whitespace in the value submitted from the Redactor field. 

If I add a simple `trim()` to the `$value` checked in https://github.com/craftcms/html-field/blob/fd3d400643fccbe27583102f581640f465987a06/src/HtmlField.php#L94, it works as expected, i.e.

```php
if (in_array(\trim($value), ['<p><br></p>', '<p>&nbsp;</p>'], true)) {
   return null;
}
```

**Comment by brandonkelly** [comment]
Fixed via `craftcms/html-field` v1.0.7 and 2.0.2.
